extensions/oplogPopulator/modules/ConnectorsManager.js

```diff
@@ -204,12 +208,14 @@ class ConnectorsManager {
                 const oldConnectors = await this._getOldConnectors(oldConnectorNames);
                 this._connectors.push(...oldConnectors);
                 this._oldConnectors.push(...oldConnectors);
+                this._metricsHandler.onConnectorsInstantiated(true, oldConnectors.length);
```

Comment:
onConnectorsInstantiated() catches the error inside the function if `this.connectors.inc` failes, but doesn't it matter for the following logic of  `this._metricsHandler.onConnectorsInstantiated(true, oldConnectors.length);` even if `this.connectors.inc` fails?

Reply:
It does affect it, the value of the metric will be false.
We don't however want to crash the oplog populator here as that will lead to some events being missed in case one or more bucket configurations change.